Contrib/NP_Score/npscorer.py

```diff
@@ -21,56 +21,70 @@
 
 
 def readNPModel(filename=os.path.join(os.path.dirname(__file__), 'publicnp.model.gz')):
-  sys.stderr.write("reading NP model ...\n")
-  fscore = pickle.load(gzip.open(filename))
-  sys.stderr.write("model in\n")
-  return fscore
+    sys.stderr.write("reading NP model ...\n")
+    fscore = pickle.load(gzip.open(filename))
+    sys.stderr.write("model in\n")
+    return fscore
 
 
-def scoreMol(mol, fscore):
-  if mol is None:
-    raise ValueError('invalid molecule')
-  fp = rdMolDescriptors.GetMorganFingerprint(mol, 2)
-  bits = fp.GetNonzeroElements()
+def scoreMolWConfidence(mol, fscore):
+    """Next to the NP Likeness Score, this function outputs a confidence value
+    between 0..1 that descibes how many fragments of the tested molecule
+    were found in the model data set (1: all fragments were found).
+
+    Returns tuple(NP Likeness, Confidnce)"""
+
+    if mol is None:
+        raise ValueError('invalid molecule')
+    fp = rdMolDescriptors.GetMorganFingerprint(mol, 2)
+    bits = fp.GetNonzeroElements()
 
-  # calculating the score
-  score = 0.
-  for bit in bits:
-    score += fscore.get(bit, 0)
-  score /= float(mol.GetNumAtoms())
+    # calculating the score
+    score = 0.0
+    bits_found = 0
+    for bit in bits:
+        if bit in fscore:
+            bits_found += 1
+            score += fscore[bit]
 
-  # preventing score explosion for exotic molecules
-  if score > 4:
-    score = 4. + math.log10(score - 4. + 1.)
-  if score < -4:
-    score = -4. - math.log10(-4. - score + 1.)
-  return score
+    score /= float(mol.GetNumAtoms())
+    confidence = round(float(bits_found / len(bits)), 3)
 
+    # preventing score explosion for exotic molecules
+    if score > 4:
+        score = 4. + math.log10(score - 4. + 1.)
+    if score < -4:
+        score = -4. - math.log10(-4. - score + 1.)
+    return score, confidence
+
+
+def scoreMol(mol, fscore):
```

Comment:
As you already started to add a comment above, why not add comment for the other methods too.